### ISSUE TITLE
Strimzi Operator does not conform to Kafka Connect API implementation

### ISSUE THREAD

**Issue by John-Athan** [strimzi/strimzi-kafka-operator#5427]
#### Description
First of all: You guys have a great product, we have been using it for over a year now and it works like a charm :) 
About the issue:

Since Version 2.5, Kafka Connect exposes an Endpoint for getting all topics related to a Connector via its REST API  (see [the original KIP](https://cwiki.apache.org/confluence/display/KAFKA/KIP-558%3A+Track+the+set+of+actively+used+topics+by+connectors+in+Kafka+Connect))
While the original KIP proposed the Response Payload to look as follows:
```
{
  "some-source": {
    "topics": [
      "foo",
      "bar",
      "baz",               
    ]
  }
}
```
The documentation by Confluent states the same: https://docs.confluent.io/platform/current/connect/references/restapi.html#get--connectors-(string-name)-topics
The [actual Code](https://github.com/apache/kafka/blob/2.7/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java#L198), however, produces a result of the following form:
```
{
  "some-source": {
    "connector": "some-source",
    "topics": [
      "foo",
      "bar",
      "baz",               
    ]
  }
}
```
This poses a problem to the Strimzi Operator since it expects the response to be of type `Map<String, Map<String, List<String>>>` (see [here](https://github.com/strimzi/strimzi-kafka-operator/blob/61a2301390fb9ecc87feb1925d0c2d2f1b2f8107/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java#L610)), but in Kafka Connect, the return type is actually `Map<String, Map<String, Object>>` (see [this test](https://github.com/apache/kafka/blob/2.7/connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResourceTest.java#L884), for example)

We have noticed that in some cases, the field "connector" is used. In these cases, the Strimzi Operator will fail during reconciliation with the following message:
```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `java.util.ArrayList<java.lang.Object>` out of VALUE_STRING token 
at [Source: (byte[])"{"myconnector":{"connector":"myconnector","topics":["foo", "bar"]}}"; line: 1, column: 39] (through reference chain: java.util.LinkedHashMap["myconnector"]->java.util.LinkedHashMap["connector"])
```

**Expected behavior**
The Operator should expect an object of type `Map<String, Map<String, Object>>`, as it is specified in the code of Kafka Connect.

**Environment:**
 - Strimzi version: 0.22.1
 - Installation method: Strimzi Operator via Helm chart, Kafka Connect & Connector via Strimzi's Custom Resources
 - Kubernetes cluster: Kubernetes 1.18
 - Infrastructure: Amazon EKS

**Comment by scholzj** [comment]
So, when does it produce one and when the other layout? I have never seen this error, so I assume my Connect never produced this format?

**Comment by John-Athan** [comment]
We have an older Strimzi Kafka Connect version 2.5 running, which also runs with different Connectors and diffferent Plugins. It does not produce this issue. We also have a Confluent Kafka Connect Version 6.1.2 running, which runs Confluent S3 Sink Connectors with the respective Plugins. The issues occur when we move the Confluent S3 Sinks to Strimzi Kafka Connect, therefore moving the Confluent Plugins onto the Strimzi Docker Image.

But as Kafka Connect is using a different layout than that used by Strimzi, I would consider it decoupled from any specific connector or setup.

**Comment by scholzj** [comment]
> But as Kafka Connect is using a different layout than that used by Strimzi, I would consider it decoupled from any specific connector or setup.

But this is the confusing point ... I do not see this different layout in my Connect.

**Comment by John-Athan** [comment]
I'm sorry, but I could not track down why the `connector` field is missing in most cases. The code clearly states that it should exist, as does the unit test I posted above. 
So the problem here, I think, is that it is not clear why the layout in the Kafka repository differs from Confluent's documentation as well as KIP-558. I will try to get in touch with the Kafka developers in order to sort this out.

**Comment by scholzj** [comment]
@tombentley @tomncooper Any chance you can try to have a look into the Kafka code to see why/when does the other format happen? Thanks

**Comment by John-Athan** [comment]
For reference only, this is the related issue I opened: https://issues.apache.org/jira/browse/KAFKA-13205

**Comment by scholzj** [comment]
When you say that from the code, the format should be

```json
{
  "some-source": {
    "connector": "some-source",
    "topics": [
      "foo",
      "bar",
      "baz",               
    ]
  }
}
```

Why do you think so?

It seems to be serialising `https://github.com/apache/kafka/blob/7b5db94c967fe56ef478011655aa85d7d41c8db5/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ActiveTopicsInfo.java` and I do not think that will serialize the `connector` field into the JSON.

**Comment by John-Athan** [comment]
Following your comment, I looked further into what could be the reason for this issue.
I was able to isolate the `Confluent Control Center` as the cause of this issue. If it is included in the classpath, it will cause the issue I described above. 
Without Confluent Control Center, everything is fine. For us, this solves the issue. As this is also not an issue directly related to Strimzi, I will close the issue.
Thanks @scholzj
